Simplenote/SimplenoteAppDelegate.m

```diff
@@ -473,6 +479,8 @@ - (BOOL)applicationShouldHandleReopen:(NSApplication *)sender hasVisibleWindows:
         return YES;
     }
 
+    [self authenticateSimperiumIfAccountDeletionRequested];
```

Comment:
This should probably go here:

```
- (void)applicationWillBecomeActive:(NSNotification *)notification
{
    [self authenticateSimperiumIfAccountDeletionRequested];
}
```

`shouldHandleReopen` is invoked whenever you click on the Simplenote Icon (in the Dock). Furthermore, because of the statement above, if the window was already visible, I'm afraid this code will never run.